File: project_euler/Problem003/Problem003.go
package project_euler

import "fmt"

func Problem003() {
    target := int64(600851475143)
    for 0 == (target & 1) {
        target >>= 1
    }
    if 1 == target {
        target = 2
    }

    factor := int64(3)
    for factor < target {
        for 0 == target % factor && target != factor {
            target /= factor
        }
        factor += 2
    }
    fmt.Println(target)
}

File: project_euler/Problem003/Problem003.java
package project_euler;

public class Problem003 {
    public void solution() {
        long target = 600851475143L;
        while (0 == (target & 1)) {
            target >>= 1;
        }
        if (1 == target) {
            target = 2L;
        }

        long factor = 3L;
        while (factor < target) {
            while (0 == target % factor && target != factor) {
                target /= factor;
            }
            factor += 2L;
        }
        System.out.println(target);
    }
}
